src/Aggregator/Product/ProductByTypeCountAggregator.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RunAsRoot\PrometheusExporter\Aggregator\Product;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use RunAsRoot\PrometheusExporter\Api\Data\MetricInterface;
+use RunAsRoot\PrometheusExporter\Api\MetricAggregatorInterface;
+use RunAsRoot\PrometheusExporter\Repository\MetricRepository;
+use RunAsRoot\PrometheusExporter\Service\UpdateMetricService;
+
+class ProductByTypeCountAggregator implements MetricAggregatorInterface
+{
+    private const METRIC_CODE = 'magento_products_by_type_count_total';
+
+    private MetricRepository $metricRepository;
+    private SearchCriteriaBuilder $searchCriteriaBuilder;
+    private UpdateMetricService $updateMetricService;
+    private ResourceConnection $resourceConnection;
+
+    public function __construct(
+        MetricRepository $metricRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        UpdateMetricService $updateMetricService,
+        ResourceConnection $resourceConnection
+    ) {
+        $this->metricRepository = $metricRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->updateMetricService = $updateMetricService;
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    public function getCode(): string
+    {
+        return self::METRIC_CODE;
+    }
+
+    public function getHelp(): string
+    {
+        return 'Magento 2 Product by type Count';
+    }
+
+    public function getType(): string
+    {
+        return 'gauge';
+    }
+
+    public function aggregate(): bool
+    {
+        $this->resetMetrics();
+
+        $connection = $this->resourceConnection->getConnection('products');
```

Comment:
There is no default connection with `products` name. Magento Catalog setup uses default connection, so I prefer to leave a default value here

Reply:
Done :)!